[Cognitive Services] October Preview ARM API

CarlOchs opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [ ] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

CarlOchs committed:
Base 2025-10-01-preview on 2025-09-01 stable

CarlOchs committed:
Update api-version in examples.

fmabroukmsft committed:
Add network management v2 to cog services swagger (#37793)

* Copy files from preview/2025-07-01-preview

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

* Update version to preview/2025-10-01-preview

Updated the API version from preview/2025-07-01-preview to preview/2025-10-01-preview.

* Added tag for 2025-10-01-preview in readme file

* Add NMS v2 to cogsvc swagger.

* Update parameter casing.

* Update api versions in examples.

* Update NMS parameter properties

* Copy over pre-approved suppressions from previous API version.

* Add suppressions for NMS v2 swagger additions.

* Remove unneeded properties, add provision route

* Update accountNameParameter casing.

* Update api version in swagger to 2025-10-01-preview.

* Consistent naming for package tag in readme.

* Prettier gate formatting fix.

* Retrigger pipelines

* Add summary for outbound rule operation.

---------

Co-authored-by: Mike Harder <mharder@microsoft.com>

fmabroukmsft committed:
Add agentic application to cog services swagger. (#37849)

* Add agentic application to cog services swagger.

* Address pipelines.

* Fix spellcheck pipeline.

ravimeda commented:
>       "delete": {

Curious if you support synchronous delete? If so, include a 200 (OK) response.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json:6047 in 719a515. [](commit_id = 719a515a4e8c2d06c37a827fa8f8cafce089dd0a, deletion_comment = False)

ravimeda commented:
>         "summary": "The GET API for retrieveing a single outbound rule of the managed network associated with the cognitive services account.",

Typo

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json:5370 in 719a515. [](commit_id = 719a515a4e8c2d06c37a827fa8f8cafce089dd0a, deletion_comment = False)

ravimeda reviewed (changes requested):
:clock1:

ravimeda reviewed (approved):
:shipit:

aditya30394 committed:
Add deployment pause/resume APIs and service tier management (#37935)

* feat(cognitiveservices): Add deployment pause/resume APIs and service tier management

- Add serviceTier enum property to DeploymentProperties (Default, Priority)
- Add deploymentState enum property to DeploymentProperties (Running, Paused)
- Add POST /deployments/{deploymentName}/pause endpoint for pausing deployments
- Add POST /deployments/{deploymentName}/resume endpoint for resuming deployments
- Create API examples for pause and resume operations
- Update existing deployment examples to include new properties
- Add comprehensive descriptions with usage guidance and SKU restrictions
- Document HTTP 423 behavior for paused deployments
- Mark both operations as idempotent

This enables customers to control deployment processing priority through
service tiers and pause/resume deployment inferencing for cost optimization
and resource management.

* need to add pattern for lint errors

* deploymentNameForActionParameter

vermaanjali4 committed:
Update API spec for the Safety provider Get and List APIs (#37525)

* Update API spec for the Safety provider Get and List APIs
* Add webhook URL to the spec
* fix duplicate error
* address NDA issue
* update safetyprovider urls to subscription level
* add susbscription level safety provider
* Add test connection and sub rai policy
* remove unused changes
* update format
* Update testRaiExternalSafetyProvider endpoint path
* update policy name parameter in subscription level
* Club together the tags for all safety provider apis
* fix prettier error
* Fix swagger error
* Fix validation errors
* Add suppressions

---------

Co-authored-by: tparuchuri <tparuchuri@example.com>
Co-authored-by: Tejaswi Paruchuri <86433817+Tparuchuri@users.noreply.github.com>
Co-authored-by: CJ Ochs <carlochs@microsoft.com>

samnamdeo1 committed:
Add ToolLabel feature to cogsvc control plane (#37951)

* Adding spec json for RaiIfcRule get
* Add IfcRule and Tool Label feature to cogsvc control plane
* Update raiToolLabelNameParameter
* Update for RaiIfcRule and RaiToolLabel
* Fix required ifcRuleName
* Add suppressions for AvoidAdditionalProperties LintDiff flag
* Add "hitl" as valid word
* Update Enums for source and action in RaiPolicyContentFilter
* Remove RaiIfcRule
* Add 'hitl' to cspell.yaml dictionary
---------

Co-authored-by: CJ Ochs <carlochs@microsoft.com>
Co-authored-by: CJ Ochs <111904639+CarlOchs@users.noreply.github.com>

fmabroukmsft committed:
Address synchronous to asynchronous routes and unneeded pagination.

fareedmabrouk commented:
> > ```
> >       "200": {
> > ```
> 
> is this a synchronus delete?
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json:5332 in [5290c6c](https://github.com/Azure/azure-rest-api-specs/commit/5290c6c43411627272160fc0d85adad111328353). [](commit_id = [5290c6c](https://github.com/Azure/azure-rest-api-specs/commit/5290c6c43411627272160fc0d85adad111328353), deletion_comment = False)

Addressed

fareedmabrouk commented:
> > ```
> >     "x-ms-pageable": {
> > ```
> 
> this for a post?
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json:6285 in [5290c6c](https://github.com/Azure/azure-rest-api-specs/commit/5290c6c43411627272160fc0d85adad111328353). [](commit_id = [5290c6c](https://github.com/Azure/azure-rest-api-specs/commit/5290c6c43411627272160fc0d85adad111328353), deletion_comment = False)

Addressed

psah434 commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,109 @@ These are the global settings for the CognitiveServices API.
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{name}"].get.responses.200.schema
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{name}"].put.responses.200.schema
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{name}"].put.responses.201.schema
+  - code:  ProvisioningStateMustBeReadOnly
```
same as above

psah434 replied:
can this be more specific

fmabroukmsft replied:
same as above, did you have a different path in mind? it's already applied to each route individually where necessary

psah434 commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,109 @@ These are the global settings for the CognitiveServices API.
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{appName}/deployments/{deploymentName}"].get.responses.200.schema
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{appName}/deployments/{deploymentName}"].put.responses.200.schema
+      - $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/projects/{projectName}/applications/{appName}/deployments/{deploymentName}"].put.responses.201.schema
+  - code: ArmResourcePropertiesBag
```
can you elaborate?

fmabroukmsft commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,109 @@ These are the global settings for the CognitiveServices API.
+input-file:
+  - Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json
+suppressions:
+  - code:  ProvisioningStateMustBeReadOnly
```
this one is necessary, it's for the new applications routes that trigger the lintdiff check for provisioningstate must be readonly despite already having it set.

fmabroukmsft committed:
Sync connections in cog services swagger (#38250)

* Correct connection categories

* Update auth types
